homeassistant/components/withings/config_flow.py

```diff
@@ -0,0 +1,242 @@
+    }
+
+
+@config_entries.HANDLERS.register(const.DOMAIN)
+class WithingsFlowHandler(config_entries.ConfigFlow):
+    """Handle a config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialize flow."""
+        self.flow_profile = None
+
+    def async_profile_config_entry(
+            self,
+            profile: str
+    ) -> Optional[ConfigEntry]:
+        """Get a profile config entry."""
+        entries = self.hass.config_entries.async_entries(const.DOMAIN)
+        for entry in entries:
+            if entry.title == profile:
+                return entry
+
+        return None
+
+    def get_auth_client(self, profile: str):
+        """Get a new auth client."""
+        flow = self.hass.data[DATA_FLOW_IMPL][profile]
+        client_id = flow[const.CLIENT_ID]
+        client_secret = flow[const.CLIENT_SECRET]
+        base_url = flow[const.BASE_URL].rstrip('/')
+        callback_path = auth_callback_path(profile).lstrip('/')
+        auth_callback_path_str = const.AUTH_CALLBACK_PATH.lstrip('/')
+
+        # Clean up the base url in case the user configured a bit too much.
+        if base_url.endswith(callback_path):
+            base_url = base_url[:-len(callback_path)]
+        if base_url.endswith(auth_callback_path_str):
+            base_url = base_url[:-len(auth_callback_path_str)]
+
+        callback_uri = '{}/{}'.format(
+            base_url.rstrip('/'),
+            callback_path.lstrip('/')
+        )
+
+        return nokia.NokiaAuth(
+            client_id,
+            client_secret,
+            callback_uri,
+            scope=','.join(['user.info', 'user.metrics', 'user.activity'])
+        )
+
+    async def async_step_user(self, user_input=None):
+        """Create an entry for selecting a profile."""
+        flows = self.hass.data.get(DATA_FLOW_IMPL, {})
+
+        if not flows:
+            _LOGGER.debug("no flows")
+            return self.async_abort(reason='no_flows')
+
+        if user_input and const.PROFILE in user_input:
+            return await self.async_step_auth(user_input)
+
+        return self.async_show_form(
+            step_id='user',
+            data_schema=vol.Schema({
+                vol.Required('profile'):
+                    vol.In(list(flows))
+            }))
+
+    async def async_step_auth(self, user_input=None):
+        """Create an entry for auth."""
+        if user_input and const.CODE in user_input:
+            return await self.async_step_code(user_input)
+
+        errors = {}
+
+        if user_input is not None:
+            errors['base'] = 'follow_link'
+
+        profile = user_input[const.PROFILE]
+
+        auth_client = self.get_auth_client(profile)
+
+        self.hass.http.register_view(WithingsAuthCallbackView(
+            self.flow_id,
+            profile
+        ))
+
+        url = auth_client.get_authorize_url()
+
+        return self.async_show_form(
```

Comment:
We should use `self.async_external_step`. For an example, see https://github.com/home-assistant/home-assistant/pull/23782